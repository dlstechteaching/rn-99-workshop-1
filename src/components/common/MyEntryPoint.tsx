import React from 'react';
import { TouchableOpacity, StyleSheet, View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface MyEntryPointProps {
    iconName: string;
    title: string;
    desc?: string;
}

export const MyEntryPoint: React.VFC<MyEntryPointProps> = ({ iconName, title, desc }) => {
	return (
		<View style={styles.container}>
			<View style={styles.main}>
				<View style={styles.icon}>
					<Ionicons name={iconName as any} size={18} color="green"/>
				</View>
				<View style={styles.text}>
					<Text>{title}</Text>
					{ desc && <Text style={styles.desc}>{desc}</Text>}
				</View>
			</View>
			<TouchableOpacity style={styles.button}>
				<Ionicons name={'md-arrow-forward'} size={14} color="green" />
			</TouchableOpacity>
		</View>
	); 
};

const styles = StyleSheet.create({
	container: {
		backgroundColor: 'white',
		padding: 10,
		borderRadius: 10,
		justifyContent: 'space-between',
		flexDirection: 'row',
		marginBottom: 7,
		alignItems: 'center'
	},
	main: {
		width: '70%',
		flexDirection: 'row',
		alignItems: 'center'
	},
	icon: {
		backgroundColor: 'rgb(253, 246, 239)',
		padding: 5,
		borderRadius: 12
	},
	text: {
		marginLeft: 10
	},
	desc : {
		color: 'rgb(152, 152, 152)'
	},
	button: {
		borderRadius: 25,
		flexShrink: 1,
		backgroundColor: 'rgb(236, 248, 248)',
		height: 35,
		width: 35,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center'
	},
});