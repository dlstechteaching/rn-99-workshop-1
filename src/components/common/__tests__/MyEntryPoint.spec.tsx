import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { MyEntryPoint } from "../MyEntryPoint";

describe("MyEntryPoint tests", () => {
  let component: ShallowWrapper;

  it("should render properly", () => {
    givenComponent();
    thenItRendersProperly();
  });

  function givenComponent() {
    component = shallow(
      <MyEntryPoint 
        iconName={'label'} 
        title={'title'} 
        desc={'desc'} />
    );
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }

});