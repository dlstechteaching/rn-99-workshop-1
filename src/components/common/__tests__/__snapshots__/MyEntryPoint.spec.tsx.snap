// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEntryPoint tests should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 7,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "70%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(253, 246, 239)",
          "borderRadius": 12,
          "padding": 5,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="green"
        name="label"
        size={18}
      />
    </View>
    <View
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      <Text>
        title
      </Text>
      <Text
        style={
          Object {
            "color": "rgb(152, 152, 152)",
          }
        }
      >
        desc
      </Text>
    </View>
  </View>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(236, 248, 248)",
        "borderRadius": 25,
        "display": "flex",
        "flexShrink": 1,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="green"
      name="md-arrow-forward"
      size={14}
    />
  </ForwardRef>
</View>
`;
